developer-docs-site/docs/guides/nfts/mint-nft-v2-cli.md

```diff
@@ -0,0 +1,918 @@
+
+This means that there are no accounts initialized anywhere, the time on-chain hasn't been set, and that you need to set all these things up when you begin your tests. We'll write a helper function that we call in each of our unit tests that initializes our testing environment.
+
+Note that when you see `#[test_only]` above a function, it means the function is a function that can only be called in the test environment. `#[test]` marks a function as a unit test.
+
+```rust
+// dependencies only used in test, if we link without #[test_only], the compiler will warn us
+#[test_only]
+use aptos_std::token_objects::collection::{Self, Collection};
+#[test_only]
+use aptos_std::token_objects::aptos_token::{Self};
+// ...etc
+
+
+#[test_only]
+fun setup_test(
+    owner: &signer,
+    new_admin: &signer,
+    nft_receiver: &signer,
+    nft_receiver2: &signer,
+    aptos_framework: &signer,
+    timestamp: u64,
+) acquires MintConfiguration {
+    timestamp::set_time_has_started_for_testing(aptos_framework);
+    timestamp::update_global_time_for_test_secs(timestamp);
+    account::create_account_for_test(signer::address_of(owner));
+    account::create_account_for_test(signer::address_of(nft_receiver));
+    account::create_account_for_test(signer::address_of(nft_receiver2));
+    account::create_account_for_test(signer::address_of(aptos_framework));
+    account::create_account_for_test(signer::address_of(new_admin));
+    initialize_collection(
+        owner,
+        get_collection_name(),
+        get_collection_uri(),
+        MAXIMUM_SUPPLY,
+        ROYALTY_NUMERATOR,
+        ROYALTY_DENOMINATOR,
+        get_token_name(),
+        get_token_uri(),
+    );
+}
+
+// Helper functions for the default values we've been using.
+// We use these to avoid `utf8` casts, since we can't set `String` type const variables.
+#[test_only]
+const COLLECTION_NAME: vector<u8> = b"Krazy Kangaroos";
+#[test_only]
+public fun get_collection_name(): String { string::utf8(COLLECTION_NAME) }
+// ...etc
+```
+We initialize the time on-chain, set it to `timestamp`, and then create accounts for all of our test accounts. Then we initialize the collection, since it's used in all of our test functions.
+
+Now let's write our happy path. This tests that all the expected functionality is working as intended in a scenario where nothing goes wrong.
+
+We'll write checks for our list #1-#4 above at the end of the test.
+
+In a `#[test]` function, we can specify accounts we want to name, set their address, and pass them in as signers to the function as if they had signed the transaction. For all of our tests, we're going to use the same addresses for simplicity's sake.
+
+Now let's pass them in as signers and set up our happy path test:
+
+```rust
+
+```
+
+For the sake of brevity, we'll only explain a single example of a negative test condition here. We'll test that setting a new admin results in the old admin being unable to call admin-only functions:
+
+```rust
+
+```
+
```

Comment:
hmm?

Reply:
I think I went overboard with some of the explanations here and tried to fit way too much into one PR...

I'm going to revamp a lot of this stuff (mostly by removing lots of content that will be linked in other sections of the docs) and leave the unit tests out of the tutorial explanations entirely.